common/subdevice-model.cpp

```diff
@@ -81,6 +81,18 @@ namespace rs2
             auto opt = static_cast<rs2_option>(i);
 
             opt_container[opt] = create_option_model(opt, opt_base_label, model, options, options_invalidated, error_message);
+            model->s->on_options_value_update( [model]( const options_list & list )
+            {
+                for( auto opt_id : list )
+                {
+                    auto it = model->options_metadata.find( opt_id );
+                    if( it != model->options_metadata.end() &&
+                        model->options_metadata[opt_id].endpoint->supports( opt_id ) )
+                    {
+                        model->options_metadata[opt_id].value = model->options_metadata[opt_id].endpoint->get_option( opt_id );
```

Comment:
This will do a query?

Reply:
Yes, currently will query. Hopefully we will implement `get_last_known_value` for all options in your planned refactor, then we would be able to use it and save time.

Reply:
Have you tested what happens with the power state between such queries? I.e., would it make sense to power-on the endpoint before the callbacks?

Reply:
UVC options that need it set the power state. The options-watcher cannot set it for all options.
I thought about optional before/after update callback, but even the `synthetic_sensor` does not know if it's raw sensor is `uvc_sensor` to set such callbacks.

Reply:
But how would we do it?
You could have logic (like power-on/-off) before/after we actually retrieve options. Maybe thru a callback? The owner knows what it needs, right?

Reply:
That is what I meant . Currently `options-watcher` owner is a `syntethic_sensor`. It has a raw sensor, but it does not know if it is a `uvc_sensor` or not.
Options that are `uvc_xu_option` or `uvc_pu_option` use have a reference to a `uvc_sensor` and `invoke_powered` internally when setting or querying. The `syntethic_sensor` doesn't know if an option is UVC when it registers it.

Reply:
Repeatedly powering on then off is time-consuming.
We can power-on before doing lots of queries, then power-off when done.
We have other examples in the code that do this (Nir added some code like this, to speed up device initialization).

What I'm asking is if should add a mechanism where the owner of the watcher can say "do this before updating" and "do this after". That way the watcher would be flexible.

I guess with the current virtual `update_options()`, this can be done with a `synthetic_options_watcher` that overrides and calls the base class. This might be worth doing? Maybe check with Nir?